lib/payment_processor/clients/braintree/transaction.rb

```diff
@@ -30,19 +30,8 @@ def sale
               email: @user[:email]
             }
           )
-          process_response(transaction)
         end
-
-        def process_response(braintree_response)
-          puts "processing response"
-          if braintree_response.success?
-            { success: true, transaction_id: braintree_response.transaction.id }.to_json
-          else
-            { success: false, errors: [] }.to_json
-          end
-        end
-
       end
     end
   end
```

Comment:
When calling `#sale` I think it would be nice to get back an instance of `::Braintree::Transaction`, because of the amount of data it holds. The the calling object (in our case, a controller) loses all that information, and instead gets a short piece of json: `{ success: true, transaction_id: braintree_response.transaction.id }`. 

I agree that we could have another class for better presenting the object as a view (decorator pattern), so we can cut out any logic from our jbuilder template (view code).

I'm not sure if I've answered your question, so shout if I haven't.